src/cpp/src/lora_common.hpp

```diff
@@ -18,13 +18,14 @@ namespace utils {
 
 template <typename T>
 struct LoRAParts {
-    T alpha, A, B;
+    T alpha, A, B, weight;
```

Comment:
Technically, `weight` is not a LoRA, so I would suggest to consider representing separately a normal LoRA triplet (`A`, `B`, and `alpha`) and `weights` at a higher level: where we obtain grouped tensors from `safetensors` file by extending the `value` type in that map. Otherwise, adding `weight` here, in the basic template, will leak it everywhere.